### ISSUE TITLE
Compilation fails due to a (documented) bug in axios

### ISSUE THREAD

**Issue by dklischies** [nestjs/nest#5611]
## Potential Commit/PR that introduced the regression**

https://github.com/nestjs/nest/commit/b2d1d464460304b1eeeb4edbc438d513346b4c0c

## Describe the regression

A regression in axios prevents arbitrary Nest applications from building.

```
Starting compilation in watch mode...

node_modules/@nestjs/common/node_modules/axios/index.d.ts:62:38 - error TS2304: Cannot find name 'ProgressEvent'.
62   onUploadProgress?: (progressEvent: ProgressEvent) => void;
                                        ~~~~~~~~~~~~~
node_modules/@nestjs/common/node_modules/axios/index.d.ts:63:40 - error TS2304: Cannot find name 'ProgressEvent'.
63   onDownloadProgress?: (progressEvent: ProgressEvent) => void;
                                          ~~~~~~~~~~~~~
```

The issue was already fixed https://github.com/axios/axios/issues/3219, so all that is required is bumping the Axios version to 0.21.0.

Yarn users can add the following block to their package.json, which serves as a workaround and validates updating the dependency version indeed fixes the issue:

```json
    "resolutions": {
        "@nestjs/common/axios": "0.21.0"
    }
```

## Input Code

This is not dependent on application code. See below for an affected tsconfig.json. In particular, please note that we do not specify nor polyfill any DOM libraries, which triggers the dependency's issue. 

## Environment
<pre><code>
 "@nestjs/common": "^7.4.4"
</code></pre>

tsconfig

```json
{
    "compilerOptions": {
        "module": "commonjs",
        "declaration": true,
        "noImplicitAny": false,
        "removeComments": true,
        "lib": ["es2019", "es2020.bigint", "es2020.string", "es2020.symbol.wellknown"],
        "emitDecoratorMetadata": true,
        "experimentalDecorators": true,
        "target": "es2019",
        "sourceMap": true,
        "outDir": "./dist",
        "baseUrl": "./"
    },
    "exclude": ["node_modules", "dist"]
}
```

**Comment by jmcdo29** [comment]
Good callout. We'll work on getting the dependency updated soon.

**Comment by kamilmysliwiec** [comment]
Fixed in 7.5.0
